File: lib/types/input.ts
/**
 * 백엔드 `src/patent_agent/models/input.py`와 1:1 대응.
 */

export interface PatentDoc {
  application_number: string;
  title: string | null;
  abstract: string | null;
  claims_text: string;
  spec_text: string | null;
  filing_date: string | null;
}

export interface PriorArtDoc {
  cited_art_id: string;
  document_number: string;
  title: string | null;
  abstract: string | null;
  claims_text: string | null;
  full_text: string | null;
}

export interface OfficeActionRaw {
  application_number: string;
  raw_text: string;
  received_date: string | null;
}

File: backend/src/patent_agent/models/input.py
from pydantic import BaseModel


class PatentDoc(BaseModel):
    application_number: str
    title: str
    abstract: str
    claims: dict[int, str]            # {1: "웨트 마스터 배치...", ...}
    spec_paragraphs: dict[str, str]   # {"0001": "본 발명은...", ...}


class PriorArtDoc(PatentDoc):
    prior_art_id: str                 # "인용발명1"
    publication_number: str


class OfficeActionRaw(BaseModel):
    application_number: str
    raw_dict: dict
